src/actions/preview.js

```diff
@@ -20,14 +20,54 @@ import { isEqual } from "lodash";
 import type { ThunkArgs } from "./types";
 import type { AstLocation } from "../workers/parser";
 
-const extraProps = {
-  react: { displayName: "this._reactInternalInstance.getName()" },
-  immutable: {
-    isImmutable: exp => `Immutable.Iterable.isIterable(${exp})`,
-    entries: exp => `${exp}.toJS()`,
-    type: exp => `${exp}.constructor.name`
+function isImmutableResult(preview) {
+  if (!preview || !preview.ownProperties) {
+    return false;
   }
-};
+  const { _root, __ownerID, __altered, __hash } = preview.ownProperties;
+  return !!(_root && __ownerID && __altered && __hash);
+}
+
+async function getExtraProps(client, frame, expression, result) {
+  let immutableType = null;
+  let immutableEntries = null;
+  const isImmutable = isImmutableResult(result.preview);
+  const frameId = frame.id;
+
+  const reactDisplayName = await client.evaluate(
```

Comment:
can we do the same thing and check to see if it `isReact` as well before client.evaluate